src/renderer/modules/components/KeybindItem.tsx

```diff
@@ -0,0 +1,295 @@
+import  React  from "@common/react";
+
+import { FormItem, Text } from ".";
+import { KeyboardEvent } from "electron";
+import "./KeybindItem.css";
+
+type ValueArray = Array<{
+  altKey: boolean | undefined;
+  code: string | undefined;
+  ctrlKey: boolean | undefined;
+  key: string | undefined;
+  keyCode: number | undefined;
+  metaKey: boolean | undefined;
+  shiftKey: boolean | undefined;
+}>;
+
+interface ButtonsProps {
+  isRecording: boolean;
+  toggleRecording: () => void;
+  recordedKeybind: ValueArray;
+  clearKeybind: () => void;
+}
+
+interface KeybindProps {
+  title?: string;
+  children?: string;
+  note?: string;
+  value: ValueArray;
+  placeholder?: string;
+  onChange: (newValue: ValueArray) => void;
+}
+interface KeybindState {
+  recordedKeybind: ValueArray;
+  currentlyPressed: number[];
+  isRecording: boolean;
+  isEditing: boolean;
+  timer: null | NodeJS.Timer;
+}
+
+interface ExtendedKeyboardEvent extends KeyboardEvent {
+  code: string | undefined;
+  key: string | undefined;
+  keyCode: number | undefined;
+}
+export type KeybindType = React.ComponentClass<KeybindProps, KeybindState>;
+
+const Buttons: React.FC<ButtonsProps> = (props) => {
+  if (!props.recordedKeybind.length && !props.isRecording) {
+    return (
+      <div className="buttons">
+        <div
+          className={`button ${props.isRecording ? "recording" : ""}`}
+          onClick={(event) => {
+            event.stopPropagation();
+            props.toggleRecording();
+          }}>
+          <svg
```

Comment:
Make a new component for each icon, they don't need to be exported, just to make this clear.